### ISSUE TITLE
`RKE2ControlPlane` version upgrade is causing unintended upgrade for new worker nodes

### ISSUE THREAD

**Issue by Danil-Grigorev** [rancher-sandbox/cluster-api-provider-rke2#315]
**What happened:**
Worker nodes are always provisioned with `RKE2ControlPlane` version. Version upgrades intended only for CP nodes are causing new worker nodes provisioned after that to inherit this version, regardless of the version specified in the `MachineDeployment` or `Machine` itself.

**What did you expect to happen:**
Version specified in `MachineDeployment` is the version of kubernetes+rke2 installed on the worker nodes at all times.

**How to reproduce it:**
1. Create the following `MachineDeployment` and the rest of the resources:
```yaml
apiversion: cluster.x-k8s.io/v1beta1
kind: MachineDeployment
metadata:
  name: test-md-0
spec:
  clusterName: test
  replicas: 1
  selector:
    matchLabels:
      cluster.x-k8s.io/cluster-name: test
  template:
    spec:
      version: v1.28.0+rke2r1
      clusterName: test
      bootstrap:
        configRef:
          apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
          kind: RKE2ConfigTemplate
          name: test-md-0
      infrastructureRef:
        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
        kind: DockerMachineTemplate
        name: test-md-0
```
2. Create or update RKE2ControlPlane version to a different one from `v1.28.0+rke2r1`.
3. Scale the machine deployment and check the version of the kubernetes components on the provisioned node. They will differ from `v1.28.0+rke2r1`.

Here is a result of testing this manually on docker following the steps. CP is on `1.28.1` and workers supposed to be on `1.27.0`:
```bash
# Output of docker machines
docker:////caprke2-e2e-rwbnbg-upgrade-control-plane-48frh  Running       v1.28.1  3m5s
docker:////caprke2-e2e-rwbnbg-upgrade-md-0-kg5xm-dqkds     Running       v1.27.0  3m7s

# SSH into worker node with 1.27.0 version
root@caprke2-e2e-rwbnbg-upgrade-md-0-kg5xm-dqkds:/# kubelet --version
Kubernetes v1.28.0
root@caprke2-e2e-rwbnbg-upgrade-md-0-kg5xm-dqkds:/# /usr/local/bin/rke2 --version
rke2 version v1.28.1+rke2r1 (4cc154f0e632a399094bb9843175f66670242ad6)
go version go1.20.7 X:boringcrypto
```

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- rke provider version:
- OS (e.g. from `/etc/os-release`):

**Comment by Danil-Grigorev** [comment]
cc @richardcase I think this is important, as it affects all deployment models, not only airGapped as per https://github.com/rancher-sandbox/cluster-api-provider-rke2/issues/297

**Comment by alexander-demicev** [comment]
@Danil-Grigorev was this fixed by merging #313 ?

**Comment by Danil-Grigorev** [comment]
Yes, it was, fixed by #313
